runtime/jsMain/src/kotlinx/serialization/DynamicObjectSerializer.kt

```diff
@@ -0,0 +1,182 @@
+package kotlinx.serialization
+
+import kotlinx.serialization.internal.StringDescriptor
+import kotlinx.serialization.internal.nullable
+import kotlinx.serialization.json.internal.BEGIN_LIST
+import kotlinx.serialization.json.internal.BEGIN_OBJ
+import kotlinx.serialization.json.internal.END_LIST
+import kotlinx.serialization.json.internal.END_OBJ
+import kotlinx.serialization.modules.EmptyModule
+import kotlinx.serialization.modules.SerialModule
+import kotlinx.serialization.modules.getContextualOrDefault
+
+
+/**
+ * Converts Kotlin data structures to plain Javascript objects
+ *
+ *
+ * Limitations:
+ * * Map keys must be of primitive or enum type
+ * * Custom serializers are not supported
+ * * Enums are serialized as the value of `@SerialName` if present or their name, in that order. Ordinals are not supported.
+ * * There is no analogue to DynamicObjectParsers' useArrayPolymorphism flag
+ *
```

Comment:
It also may be useful to put code sample here from runtime_usage.md, how it is done for JSON: https://github.com/kotlin/kotlinx.serialization/blob/85c0b12df4d7656a81bd4997b1b0a21ab91da35b/runtime/commonMain/src/kotlinx/serialization/json/Json.kt#L25